### PR TITLE
API: series query

### PR DESCRIPTION
**Pull request by rmakulov**
Test trying to insert samples every day between 1969 and 2110 with some randomization

### CODE REVIEW

File: src/test/java/com/axibase/tsd/api/method/series/SeriesQueryTest.java [hunk 1]
```diff
@@ -224,6 +230,73 @@ public void testDateFilterRangeIntersectsStorableRangeEnding() throws Exception
...
+            assertTrue("Attempt to insert date before min storable date doesn't return error",
+                    response.readEntity(String.class).startsWith("{\"error\":\"IllegalArgumentException: Too large timestamp"));
+
+            setRandomTimeDuringNextDay(calendar);
+        }
+    }
+
+    private void setRandomTimeDuringNextDay(Calendar calendar) {
+        calendar.add(Calendar.DAY_OF_YEAR, 1);
+        calendar.set(Calendar.HOUR_OF_DAY, (int) (Math.random() * 24));
```
Comment (unrealwork) [inline src/test/java/com/axibase/tsd/api/method/series/SeriesQueryTest.java]: How to reproduce this data?

Comment (rmakulov) [inline src/test/java/com/axibase/tsd/api/method/series/SeriesQueryTest.java]: For instance from logs

Comment (unrealwork) [inline src/test/java/com/axibase/tsd/api/method/series/SeriesQueryTest.java]: It's bad practiсe for unit-tests. It's more like for auto-tests

**Comment by unrealwork** [comment]
ok

File: src/test/java/com/axibase/tsd/api/method/series/SeriesQueryTest.java [hunk 2]
```diff
@@ -224,6 +230,73 @@ public void testDateFilterRangeIntersectsStorableRangeEnding() throws Exception
...
+
+        calendar.setTime(parseDate("1969-01-01T00:00:00.000Z"));
+        Date endDate = parseDate(MIN_STORABLE_DATE);
+
+        while (calendar.getTime().before(endDate)) {
+            series.setData(Collections.singletonList(new Sample(ISOFormat(calendar.getTime()), v)));
+            Response response = insertSeries(series);
+
+            assertEquals("Attempt to insert date before min storable date doesn't return error",
+                    response.getStatusInfo().getStatusCode(), Response.Status.BAD_REQUEST.getStatusCode());
```
Comment (Sild) [inline src/test/java/com/axibase/tsd/api/method/series/SeriesQueryTest.java]: is the order of expected\given correct?

File: src/test/java/com/axibase/tsd/api/method/series/SeriesQueryTest.java [hunk 3]
```diff
@@ -224,6 +230,73 @@ public void testDateFilterRangeIntersectsStorableRangeEnding() throws Exception
...
+
+        calendar.setTime(parseDate(addOneMS(MAX_STORABLE_DATE)));
+        Date endDate = parseDate("2110-01-01T00:00:00.000Z");
+
+        while (calendar.getTime().before(endDate)) {
+            series.setData(Collections.singletonList(new Sample(ISOFormat(calendar.getTime()), v)));
+            Response response = insertSeries(series);
+
+            assertEquals("Attempt to insert date before min storable date doesn't return error",
+                    response.getStatusInfo().getStatusCode(), Response.Status.BAD_REQUEST.getStatusCode());
```
Comment (Sild) [inline src/test/java/com/axibase/tsd/api/method/series/SeriesQueryTest.java]: order?

**Pull request merged**